modules/azure/src/test/java/org/testcontainers/containers/CosmosDBEmulatorContainerTest.java

```diff
@@ -0,0 +1,56 @@
+package org.testcontainers.containers;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.models.CosmosContainerResponse;
+import com.azure.cosmos.models.CosmosDatabaseResponse;
+import java.io.FileOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.testcontainers.utility.DockerImageName;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CosmosDBEmulatorContainerTest {
+
+    @Rule
+    // emulatorContainer {
+    public CosmosDBEmulatorContainer emulator = new CosmosDBEmulatorContainer(
+        DockerImageName.parse("mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator:2.11.13-c5c131e3")
+    );
+    // }
+
+    @Test
+    public void testSimple() throws Exception {
+    // buildAndSaveNewKeyStore {
+        Path keyStoreFile = Files.createTempFile("azure-cosmos-emulator", ".keystore");
+        KeyStore keyStore = emulator.buildNewKeyStore("/tmp/cosmos/appdata/sslcert.pfx");
```

Comment:
What was the reason to let the user provide the internal cert path? (as opposed to just having `emulator.getKeyStore()`)

I think we have an abstraction leak here - `/tmp/cosmos/appdata/sslcert.pfx` is something that `CosmosDBEmulatorContainer` should encapsulate, as it is specific to how the image is implemented.

Reply:
Certificate file name or generally certificate file path can be different in image versions. For example, they changed it to "default.sslcert.pfx" in "latest" docker image tag but there is no stable version tag yet.

IMO this uncertainty is not container's problem. User can provide it?

Reply:
Looks like there is an endpoint to obtain the certificate:
https://docs.microsoft.com/en-us/azure/cosmos-db/linux-emulator?tabs=ssl-netstd21#install-certificate

Let's change the impl to use that endpoint to obtain the certificate, as this must work with any version of the emulator

Reply:
This was the previous implementation in the PR, that added the ignore SSL code, which I would have considered far worse.

Reply:
@kiview not sure I follow - why can't we use the endpoint instead of copying the file? :)

Reply:
Yeah, we need to bypass SSL again if we get certificate from endpoint. At first endpoint is looking cool but  copyFileFromContainer is better than having these workaround codes IMO.

Reply:
Because the endpoint is only reachable via HTTPS (which was the original issue at [azure-sdk](https://github.com/Azure/azure-sdk-for-java/pull/22990)). So obtaining it required to disable SSL verification in the JVM, which led to a quite complex PR in the first place, using a custom `TrustManager`.

Reply:
We don't need to disable the SSL verification globally in the JVM to obtain the certificate from that http endpoint - just instantiate an http client of choice inside `CosmosDBEmulatorContainer` to download it.

Reply:
@bsideup you can check this in previous commits, I removed URL download and introduced copyFileFromContainer

https://github.com/testcontainers/testcontainers-java/pull/4303/commits/abf67b5c7619b8fa0b069ab953b09adee8dd6ba4

Reply:
Let's continue to discuss on new commit.